src/server/controllers/participants.js

```diff
@@ -223,6 +223,22 @@ exports.deleteCourse = (req, res) => {
   }
 };
 
+// Add participant Date
+exports.addDate = async (req, res) => {
+  try {
+    const participantId = req.params.id;
+    const { selectedName, date, note } = req.body;
+    await dates.create({
+      date, note, worker_name: selectedName, participant_id: participantId,
```

Comment:
Hey @mohannadhanafi I just mentioned you in a comment on another PR, I'll mention it here too - I think you might be able to avoid some bugs if you change the dates.belongsTo(workers) association to use the default targetKey (id) instead of username as a targetKey.  But that would mean that you have to change this code to use ids instead of usernames for the workers.

Reply:
I changed this by take all of object from front end
and add the id to it like this <code>dateData.participant_id = participantId;</code>
I think this problem is solved now :smile:

Reply:
I think we might be talking about different things :)

The potential problem I was describing originates in the dates.belongsTo() association here:
https://github.com/FACG5/BounceBack/blob/0582cb709dda1541e3073080d973f732ddf1320c/src/server/database/models/index.js#L26
Because of the association using `targetKey: username`, that means that if a worker model's username were ever edited, then all of the dates that had been created with that worker would be pointing at the old username.  By using a model's id as the targetKey (which is the default), that means you can edit other fields on a model without breaking existing associations that are targeting that model.

It's not actually a problem with this pull request, but if you were to go and fix the problem (by changing the targetKey), then you would have to also adjust the code in this PR (since it passes a worker username, not a worker id, to the dates create endpoint.  So there's no change required in this PR right now, just changes needed if the targetKey is changed :)